src/App/Pages/Vault/BaseCipherViewModel.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Threading.Tasks;
+using Bit.App.Abstractions;
+using Bit.App.Resources;
+using Bit.Core.Abstractions;
+using Bit.Core.Exceptions;
+using Bit.Core.Models.View;
+using Bit.Core.Utilities;
+using Xamarin.CommunityToolkit.ObjectModel;
+
+namespace Bit.App.Pages
+{
+    public abstract class BaseCipherViewModel : BaseViewModel
+    {
+        private readonly IAuditService _auditService;
+        protected readonly IDeviceActionService _deviceActionService;
+        protected readonly ILogger _logger;
+        protected readonly IPlatformUtilsService _platformUtilsService;
+
+        public abstract CipherView Cipher { get; set; }
+        public AsyncCommand CheckPasswordCommand { get; set; }
+
+        public BaseCipherViewModel()
+        {
+            _deviceActionService = ServiceContainer.Resolve<IDeviceActionService>("deviceActionService");
+            _platformUtilsService = ServiceContainer.Resolve<IPlatformUtilsService>("platformUtilsService");
+            _auditService = ServiceContainer.Resolve<IAuditService>("auditService");
+            _logger = ServiceContainer.Resolve<ILogger>("logger");
+
+            CheckPasswordCommand = new AsyncCommand(CheckPasswordAsync, allowsMultipleExecutions: false);
+        }
+
+        protected async Task CheckPasswordAsync()
+        {
+            try
+            {
+                if (string.IsNullOrWhiteSpace(Cipher.Login?.Password))
```

Comment:
Just in case, I'd check if Cipher is null: 
`string.IsNullOrWhiteSpace(Cipher?.Login?.Password)`